Ports Gems from Shiptest while giving them new features

GraveHat opened the pull request:
Ports in the gems inside shiptest-ss13/Shiptest#285

# Document the changes in your pull request

Ports over the gem functionality from Shiptest. Aka costing a lot of money and dropping from god knows where. I've made many changes on where they come from and how they cost to at least attempt to try and make them not useless or overlooked by miners like the useless artifacts they leave around.

The added features I gave to the gems are the ability to be scanned by mining scanners for free points and the ability for some of them to be welded to extract mineral sheets if available. Dropping from most megafauna and increasing their prices accordingly, the only exception to this being the rupee, frost diamond, and calcified auric which are uncommon drops from the goldgrub icewing watchers and magmawing watchers respectively. Also gave the unused Dark Cube antimagic just because it's a goat king drop.

Why should this be a thing? Well, I noticed that next to no miners know or care that the artifacts they leave inside mining sit there until the end of the round when they could have just given them away or given them to cargo so they can get credits. I hope adding an item that is quite literally free credits would get them to notice that "Hey, If I can sell this, why not sell this other thing I'm not using?"

This has been tested and is undergoing testing. Basic features work but give me a heads-up if you want to see anything changed. Here is a checklist of things I might need to do. The framework is basically finished.

Edit: Nerfs the Megafauna Cash Scale from a 5 to a 2 since no one knows it exists.
Edit 2: Updated the contents of this comment to actually match what is in the PR

- [x] Get lore director approval on these descriptions
- [x] Adjust stats to acceptable levels
- [ ] Optional: Hold off giving mining points until sold
- [ ] Optional: Notify supply when megafauna are killed and the credits they gain from them
- [x] Optional: Get better sprites if orca no likes the ones from shiptest

# Spriting
![image](https://user-images.githubusercontent.com/107460718/186411187-1e58c5f0-5cd8-4951-8700-9d6828009c01.png)
Plant for scale.

# Wiki Documentation
Stat Blocks for the gems.

Ruperium Crystal:
Export Price = 3300 Credits
Bounty Price = 10000 Credits
Materials = 20000 Uranium
Sheets = 10 Sheets of Uranium on weld
Points = 300
Drops = 10% chance on death of Goldgrub
Desc: A radioactive, crystalline compound rarely found in the goldgrubs. While able to be cut into sheets of uranium, the mineral's true value is in its resonating, humming properties, often sought out by ethereal musicians to work into their gem-encrusted instruments. As a result, they fetch a fine price in most exchanges.
![image](https://user-images.githubusercontent.com/107460718/186407405-a4dea05f-a38c-4ed5-8956-bd2c05346af6.png)

Calcified Auric:
Export Price = 4500 Credits
Bounty Price = 10000 Credits
Materials = 50000 Gold
Sheets = 25 Sheets of Gold on weld
Points = 450
Drops = 10% chance on death of Magmawing Watcher
Desc: A hot, lightly glowing mineral born from the inner workings of magmawing watchers. It is most commonly smelted down into deposits of pure gold. However, it also possesses powerful conductivity, leading some to believe it a major power component utilized by the Vxtvul Empire.
![image](https://user-images.githubusercontent.com/107460718/186404734-6f3e174e-0957-4736-8fc5-353ed27e9e4d.png)

Frost Diamond:
Export Price = 4500 Credits
Bounty Price = 10000 Credits
Materials = 30000 Diamond
Sheets = 15 Sheets of Diamond on weld
Points = 750
Drops = 10% chance on death of Icewing Watcher
Desc: A unique diamond that is produced within icewing watchers. Rarely used in traditional marriage bands, various gemstone companies now try to effect a monopoly on it, to little success. It looks like it can be cut into smaller sheets of diamond ore.
![image](https://user-images.githubusercontent.com/107460718/186407740-0a60657c-2983-49da-af5f-fe57dd7d81a7.png)

Stabilized Baroxuldium:
Export Price = 7500 Credits
Bounty Price = 15000 Credits
Materials = 80000 Plasma
Sheets = Nothing
Points = 1200
Drops = Blood Drunk Miner
Light = 2 Range, 2 Power, #62326a
Desc: A soft, glowing crystal only found in the deepest veins of plasma. Famed for its exceptional durability and uncommon beauty: widely considered to be a jackpot by mining crews. It looks like it could be destructively analyzed to extract the condensed materials within.
![image](https://user-images.githubusercontent.com/107460718/186408022-ed9ae9af-64c8-4bea-8f63-0fe229cf7001.png)

Densified Dilithium:
Export Price = 9600 Credits
Bounty Price = 18000 Credits
Materials = 64000 Dilithium
Sheets = Nothing
Points = 1600
Drops = Hierophant
Light = 2 Range, 1 Power, #b714cc
Is GPS Signal? = Harmonic
Desc: A strange mass of dilithium which pulses to a steady rhythm. Its strange surface exudes a unique radio signal detectable by GPS. It looks like it could be destructively analyzed to extract the condensed materials within.
![image](https://user-images.githubusercontent.com/107460718/186406998-6651db86-1372-4cef-81e3-01883c7be336.png)

Draconic Amber:
Export Price = 9600 Credits
Bounty Price = 18000 Credits
Materials = Nothing
Sheets = Nothing
Points = 1600
Drops = Ashdrake Necro chest
Light = 2 Range, 2 Power, #FFBF00
Desc: A brittle, strange mineral that forms when an ash drake's blood hardens after death. Cherished by gemcutters for its faint glow and unique, soft warmth. Poacher tales whisper of the dragon's strength being bestowed to one that wears a necklace of this amber, though such rumors are fictitious.
![image](https://user-images.githubusercontent.com/107460718/186408326-2e62d822-4ba4-40e0-8d77-cc62bead81cf.png)

Null Crystal:
Export Price = 10000 Credits
Bounty Price = 20000 Credits
Materials = Nothing
Sheets = Nothing
Points = 1800
Drops = Colossus Necro chest
Light = 2 Range, 1 Power, #4785a4
Size = Tiny
Desc: A shard of stellar, crystallized energy. These strange objects occasionally appear spontaneously in areas where the bluespace fabric is largely unstable. Its surface gives a light jolt to those who touch it. Despite its size, it's absurdly light.
![image](https://user-images.githubusercontent.com/107460718/186409433-0dcbe570-ea6c-4e9f-8888-f5dcce6f1395.png)

Ichorium:
Export Price = 12000 Credits
Bounty Price = 25000 Credits
Materials = Nothing
Sheets = Nothing
Points = 2000
Drops = Bubblegum Necro chest
Light = 2 Range, 3 Power, #800000
Desc: A weird, sticky substance, known to coalesce in the presence of otherwordly phenomena. While shunned by most spiritual groups, this gemstone has unique ties to the occult which find it handsomely valued by mysterious patrons.
![image](https://user-images.githubusercontent.com/107460718/186410143-2d0d26b4-5e49-41f9-8878-231b742a472b.png)

Dark Salt Lick:
Export Price = 20000 Credits
Bounty Price = 40000 Credits
Materials = Nothing
Sheets = Nothing
Points = 3000
Drops = Goat King
Light = 3 Range, 3 Power, #380a41
Is Antimagic? = Yes
Desc: An ominous cylinder that glows with an unnerving aura, seeming to hungrily draw in the space around it. The round edges of the lick are uneven patches of rough texture. Its only known property is that of anti-magic.
![image](https://user-images.githubusercontent.com/107460718/186410580-f4bb4e04-546a-4c5d-84ef-6e22b9d1fd6b.png)

# Changelog

:cl:  MemedHams, Gravehat
rscadd: Adds new Gems to Mining
rscadd: Gems are now able to be scanned with a Mining Scanner
rscadd: Some gems can be cut down for mineral sheets
rscadd: Adds in antimagic gem drop from Goat King
rscadd: Adds in gem bounties
tweak: Adds gems to mob drops
tweak: Gems are classified as lavaland exports 
imageadd: Adds the gem sprites
/:cl:

GraveHat committed:
Ports Gems from Shiptest

This also somehow involves adding in 2 new features.

Skrem7 reviewed (commented):
mmmmm new descriptions

GraveHat committed:
Apply suggestions from code review

Descriptions are Updated, still working on getting the cargo bounties and export bonus mining points.

Co-authored-by: Skrem_7 <98909416+Skrem7@users.noreply.github.com>

GraveHat committed:
Gem Bounties and Megafauna Cash Scale nerf

Adds in the gem bounties I finished 3 days ago. Also the free Cash that you get from killing megafauna gets nerfed because no one knew they existed in the first place. Also fixes some formatting.

GraveHat committed:
Price Adjustments and the Descriptions

I build.

GraveHat committed:
Names were mismatched for some reason

Metastable < Stabilized

Skrem7 reviewed (commented):
Most revising, also fixing up of typos and repetition within bounty descriptions

GraveHat committed:
Apply suggestions from code review 2

Description drop.

Co-authored-by: Skrem_7 <98909416+Skrem7@users.noreply.github.com>

cuackles reviewed (dismissed):
sprites dont really look like they would fit yogstation, a more cartoony style would probably fit better and more fitting, or something that isnt that detailed and big

GraveHat committed:
New Sprites, new gems and moving things around

Replaces the sprites of all the gems
Moves around code to be in order of magnitude
Adds in the Calcified Auric, the less valuable glowing counterpart of the Frost Diamond.
Adds in the Densified Dilithium, the signaling gem of the Hierophant.

Skrem7 reviewed (commented):
guh i meant to do this like three days ago not on merge day

cuackles reviewed (dismissed):
Again same thing forgot to remove this before

GraveHat committed:
Desc Update 3 I think?

Thanks Lore Director, you the man. I would make a thumbs up but I'm bad at emoticons.

Co-authored-by: Skrem_7 <98909416+Skrem7@users.noreply.github.com>

KillerOrcaCora reviewed (approved):
Good yk what’s not good A LIZARD NOW FOLLOWING THE RULES!!!
